### PR TITLE
[CMAKE] Add Plist creation to swiftCore

### PR DESCRIPTION
**Pull request by justice-adams-apple**
Add the Plist configuration functionality as a reusable CMAKE module and hook it up to swiftCore

### CODE REVIEW

File: Runtimes/Core/cmake/modules/Plist.cmake [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      NEWLINE_STYLE UNIX)
+  
+  set_property(TARGET ${target} APPEND PROPERTY LINK_DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/${PLIST_INFO_PLIST_OUT}")
+
+  # If Application Extensions are enabled, pass the linker flag marking
+  # the dylib as safe.
+  if (CXX_SUPPORTS_FAPPLICATION_EXTENSION AND (NOT DISABLE_APPLICATION_EXTENSION))
+    list(APPEND link_flags "-Wl,-application_extension")
+  endif()
+endfunction()
```
Comment (etcwilde) [inline Runtimes/Core/cmake/modules/Plist.cmake:32]: missing newline

File: Runtimes/Core/cmake/modules/Plist.cmake [hunk 2]
```diff
@@ -0,0 +1,32 @@
+function(generate_plist project_name project_version target)
+  if(NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
```
Comment (compnerd) [inline Runtimes/Core/cmake/modules/Plist.cmake:2]: Windows also uses plists in the toolchain. If we are adding this as a generic module, I think that Windows should also be supported.

File: Runtimes/Core/cmake/modules/Plist.cmake [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    return()
+  endif()
+  set(PLIST_INFO_PLIST "Info.plist" CACHE STRING "Plist name")
```
Comment (compnerd) [inline Runtimes/Core/cmake/modules/Plist.cmake:5]: This doesn't really help explain the cache variable. I think that we might as well as elide the description or should rework it to be more helpful.

File: Runtimes/Core/cmake/modules/Plist.cmake (hunk 2 above)
Comment (justice-adams-apple) [reply Runtimes/Core/cmake/modules/Plist.cmake]: updated to generate for all platforms but only embedd the plist on Apple platforms

**Comment by justice-adams-apple** [comment]
@swift-ci please smoke test

**Comment by justice-adams-apple** [comment]
@swift-ci please smoke test

File: Runtimes/Core/cmake/modules/Plist.cmake [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  set(PLIST_INFO_VERSION "${project_version}")
+  set(PLIST_INFO_BUILD_VERSION "${project_version}")
+
+  set(PLIST_INFO_PLIST_OUT "${PLIST_INFO_PLIST}")
+  set(PLIST_INFO_PLIST_IN "${PROJECT_SOURCE_DIR}/${PLIST_INFO_PLIST}.in")
+
+  if(APPLE)
+    target_link_options(${target} PRIVATE
+      "SHELL:-Xlinker -sectcreate -Xlinker __TEXT -Xlinker __info_plist -Xlinker ${CMAKE_CURRENT_BINARY_DIR}/${PLIST_INFO_PLIST_OUT}"
+    )
```
Comment (etcwilde) [inline Runtimes/Core/cmake/modules/Plist.cmake:16]: formatting: please put the closing parenthesis on the ending line above.
Thanks

**Review by etcwilde — APPROVED** [review]

**Comment by justice-adams-apple** [comment]
@swift-ci please smoke test Linux

**Comment by justice-adams-apple** [comment]
@swift-ci please smoke test Linux

**Pull request merged**